Sources/VLCMediaPlayer.m

```diff
@@ -219,6 +247,7 @@ @interface VLCMediaPlayer ()
     BOOL _equalizerEnabled;                     ///< Equalizer state
     libvlc_video_viewpoint_t *_viewpoint;       ///< Current viewpoint of the media
     dispatch_queue_t _libVLCBackgroundQueue;    ///< Background dispatch queue to call libvlc
+    libvlc_media_player_t *_p_mp;
```

Comment:
You are aware that the _playerinstance is playing back a stream, so you can not use the same instance for muxing later on? Or am I missing something

Reply:
I believe it would violate the state machine if you handle two player instances within a single instance of VLCMediaPlayer doing different things at the same time or consecutively. Especially event handling will be in the worst case undefined or at least hard to follow in the best case.
Following my earlier remark, I don't believe that transcoding should be part of the VLCMediaPlayer class at all, but maybe in a designated subclass.